sdk/containerregistry/azure-containerregistry/azure/containerregistry/_download_stream.py

```diff
@@ -0,0 +1,76 @@
+# coding=utf-8
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+import hashlib
+from typing import Iterator, ContextManager, TypeVar, cast, Tuple, Dict, Any
+from typing_extensions import Protocol
+
+T = TypeVar('T', bound='DownloadBlobStream')
+
+
+class GetNext(Protocol):
+    def __call__(self, *args: Any, range_header: str) -> Iterator[bytes]:
+        pass
+
+
+class DownloadBlobStream(
+    Iterator[bytes],
+    ContextManager[T],
+):
+    """Protocol for methods to provide streamed responses."""
+
+    def __init__(
+        self,
+        *,
+        response: Iterator[bytes],
+        get_next: GetNext,
```

Comment:
Surely we could have just used `Callable[]`?
Though now it will need to be `Callable[[str], Tuple[PipelineResponse, Dict[str, str]]]`

Reply:
`Callable[]` could be more flexible in case will add more parameters in future. But these two both will make mypy unhappy, mypy doesn't recognize `range_header` as one of parameters.